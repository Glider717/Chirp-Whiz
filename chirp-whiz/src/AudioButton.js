import React from 'react';

function AudioButton(props) {
    function playAudio() { 
      let x = document.getElementById("birdQAudio");
      x.play();
    }
  
      return (
        <div>
          <audio id="birdQAudio">
            <source src={props.sound} type="audio/mpeg"></source>
          </audio>
  
          <button onClick={playAudio}>{props.name}</button>
        </div>
      );
  }
  
  AudioButton.defaultProps = {
    name: "Bird Call/Sound",
    sound: ""
  }

export default AudioButton;