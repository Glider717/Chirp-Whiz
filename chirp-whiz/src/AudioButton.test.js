import React from 'react';
import ReactDOM from 'react-dom';
import AudioButton from './AudioButton';
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'

it("renders without crashing",()=>{
    const div = document.createElement("div");
    ReactDOM.render(<AudioButton />,div);
});

it('test AudioButton renders', () => {
    const container = render(<AudioButton />)
    expect(container.firstChild).toMatchSnapshot();
});