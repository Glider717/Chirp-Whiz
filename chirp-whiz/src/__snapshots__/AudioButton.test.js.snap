// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AudioButton renders 1`] = `undefined`;
